ui/src/app/workflows/components/workflow-details/artifact-panel.tsx

```diff
@@ -53,6 +53,13 @@ export const ArtifactPanel = ({
     }, [downloadUrl]);
     useCollectEvent('openedArtifactPanel');
 
+    const spec = execSpec(workflow);
+    const artifacts = spec.templates.find(t => t.name === workflow.status.nodes[artifact.nodeId].templateName)?.outputs?.artifacts;
+    let artifactGCStrategy = '';
+    if (artifacts !== undefined) {
+        artifactGCStrategy = artifacts.find(a => a.name === artifact.name).artifactGC?.strategy || spec.artifactGC?.strategy;
+    }
+
```

Comment:
Wow, you already seem like a natural in Typescript.

Reply:
I can't take all the credit for this haha - all thanks to @alexec extensive PoC.